strategy-cache-update-and-refresh/service-worker.js

```diff
@@ -0,0 +1,68 @@
+var CACHE = 'cache-update-and-refresh';
+
+// On install, cache some resource.
+self.addEventListener('install', function(evt) {
+  console.log('The service worker is being installed.');
+  // Open a cache and use `addAll()` with an array of assets to add all of them
+  // to the cache. Extend the installation until all the assets are saved.
+  evt.waitUntil(caches.open(CACHE).then(function (cache) {
+    cache.addAll([
+      './controlled.html',
+      './asset'
+    ]);
+  }));
+});
+
+// On fetch, use cache but update the entry with the latest contents
+// the from server.
+self.addEventListener('fetch', function(evt) {
+  console.log('The service worker is serving the asset.');
+  // You can use `respondWith()` to answer ASAP...
+  evt.respondWith(cacheOnly(evt.request));
+  // ...and `waitUntil()` to prevent the worker to be killed until
+  // the cache is updated.
+  evt.waitUntil(
+    update(evt.request)
+    // Finally, send a message to the referrer to inform him about the
```

Comment:
who is the referrer? why is he a man? Is it a function? then use 'it' instead. Is it a person? Then use 'they'